src/redempt/redlib/region/SpheroidRegion.java

```diff
@@ -383,7 +385,61 @@ public boolean surfaceContains(Block block) {
 		}
 		return surface.contains(block);
 	}
-	
+
+	/**
+	 * Gets a random block within a spherical region
+	 * @return Bukkit block object of the randomly picked block
+	 */
+	public Block getRandomBlock() {
+		int[] randomBlockLocation = getRandomBlockCoordinates();
+		Location pickedLocation = new Location(getWorld(), randomBlockLocation[0], randomBlockLocation[1], randomBlockLocation[2]);
+		while (!contains(pickedLocation)) {
+			pickedLocation = new Location(getWorld(), randomBlockLocation[0], randomBlockLocation[1], randomBlockLocation[2]);
+		}
+		return pickedLocation.getBlock();
+	}
+
+	/**
+	 * Gets random block within spherical region
+	 * @param random Random instance
+	 * @return Bukkit block object of the randomly picked block
+	 */
+	public Block getRandomBlock(Random random) {
+		int[] randomBlockLocation = getRandomBlockCoordinates(random);
+		Location pickedLocation = new Location(getWorld(), randomBlockLocation[0], randomBlockLocation[1], randomBlockLocation[2]);
+		while (!contains(pickedLocation)) {
+			pickedLocation = new Location(getWorld(), randomBlockLocation[0], randomBlockLocation[1], randomBlockLocation[2]);
+		}
+		return pickedLocation.getBlock();
+	}
+
+	/**
+	 * Picks a random block coordinate from a sphere region,
+	 * this treats the sphere as cuboid, be sure to discard points outside the sphere
+	 * @return Array of three integers, one for each coordinate axis
+	 */
+	private int[] getRandomBlockCoordinates(){
+		ThreadLocalRandom threadLocalRandom = getLocalRandom();
```

Comment:
Instead of copy-pasting the logic, why don't you just pass the ThreadLocalRandom to the other helper?